src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -1,6 +1,10 @@
 package edu.kis.vh.nursery;
 
-public class DefaultCountingOutRhymer {
+/**
+ * @author Marta
+ * impementacja interfejsu zamiast rozsszerzania klasy ma tę przewagę, że klasa może implementować wiele intefejsów, a rozszerzać tylko jedną klasę. 
+ */
+public class DefaultCountingOutRhymer implements IIntArrayStack {

```

Comment:
Unfortunately, it is not a proper solution.
Rhymer is not a data structure so it should not implement IIntArrayStack.
Data structures IntArrayStack and IntLinkedList should both implement the same interface (they have similar capabilities)